app/extras/queries/visible_invitable_memberships.rb

```diff
@@ -17,15 +17,19 @@ def __setobj__(obj)
   private
 
   def visible_memberships_filtered
-    Membership.where(id: visible_membership_ids)
+    Membership.select("DISTINCT ON (user_id) memberships.*")
+              .where(user_id: visible_user_ids, archived_at: nil, is_suspended: false)
               .search_for(@query)
               .limit(@limit)
   end
 
-  def visible_membership_ids
-    @visible_member_ids ||= @user.groups.flat_map { |g| g.membership_ids }
-                                        .reject   { |id| group_membership_ids.include? id }
-                                        .uniq
+  def visible_user_ids
```

Comment:
This is mildly fancy SQL, but oh-so-much faster than what we had before

It's saying
"Give me all the (distinct) user ids where if you look for a membership with that user and the given group, it doesn't exist."

This actually wants to be a query for (unique) users I think (the idea of querying invitables as memberships was a little misguided probably), but not quite ready to make that full switch just yet.